napari/layers/base/base.py

```diff
@@ -499,6 +504,10 @@ def mode(self, mode):
 
         self.events.mode(mode=str(mode))
 
+    def _get_unique_id(self):
+        """Unique id of the layer."""
+        return uuid.uuid4()
```

Comment:
Extra comment: there should be a guard here to make sure that this is only run once. I guess rolling this into the read-only property would be good!